### PR TITLE
Enhancement: Tile System BETA

### PR DESCRIPTION
**Pull request by wattsit**
### Summary
**Tile Colour Varients**
- New range of colour varients for the tiles.
- `TileSheetLoader.gd` Autoloader which shuffles and poops out tilesheets as needed.
- Removed Teal tilesheet from tileset.

**DamageTiles**
- Damage Walls added
- Damage tiles moved to a new tileset. 

zigLWdoI0n
How to place damage tiles; take note that the infection walls have left hand right hand version as well as end pieces.  

**Room Creation**
- `testing_player_spawn` node added to the CANVAS room, place this where you want the player to spawn for room scene testing.

### CODE REVIEW

File: Moonshot/Autoload/TileSheetLoader.gd [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+var Purple 	= load(tile_path + "Purple.png")
+var Red1 	= load(tile_path + "Red1.png")
+var Red2 	= load(tile_path + "Red2.png")
+var Teal1 	= load(tile_path + "Teal1.png")
+var Teal2 	= load(tile_path + "Teal2.png")
+var Yellow1 = load(tile_path + "Yellow1.png")
+var Yellow2 = load(tile_path + "Yellow2.png")
+
+var tilesheet_array = [Blue1,Blue2,Grey,Ice,Pink,Purple,Red1,Red2,Teal1,Teal2,Yellow1,Yellow2]
+var tilesheet_count = tilesheet_array.size()
```
Comment (whitew1994WW) [inline Moonshot/Autoload/TileSheetLoader.gd:19]: Can this be made into a dictionary for easy access? Not a major issue but would make more sense imho

**Review by whitew1994WW — APPROVED** [review]
Looks really good, tested and both pool and walls work locally. The only thing I would say is that if youre standing on the edge above a damage wall it damages you and pings you upwards. Not sure if this was intentional but it feels a bit wrong to me

File: Moonshot/Autoload/TileSheetLoader.gd (hunk 1 above)
Comment (wattsit) [reply Moonshot/Autoload/TileSheetLoader.gd:19]: I'll convert this to a dictionary if we plan to have set level colours. If we leave this random however I think the shuffled array is better for this.

**Comment by wattsit** [comment]
> Looks really good, tested and both pool and walls work locally. The only thing I would say is that if youre standing on the edge above a damage wall it damages you and pings you upwards. Not sure if this was intentional but it feels a bit wrong to me

Thanks for finding this, should be fixed here: https://github.com/kepler471/moonshot/pull/181/commits/41049de8fdc7441d96305c3eda5927a977883484

File: Moonshot/player/LedgeWallDetector.gd [hunk 2]
```diff
@@ -15,7 +15,7 @@ func _ready():
 	assert(ray_middle.cast_to.x >= 0)
 	assert(ray_bottom.cast_to.x >= 0)
 
-func _physics_process(delta):
+func _process(delta) -> void:
```
Comment (Jamesbarford) [inline Moonshot/player/LedgeWallDetector.gd:18]: Think this should be `_physics_process` 🤔

File: Moonshot/player/player.gd [hunk 3]
```diff
@@ -1,5 +1,6 @@
 extends KinematicBody2D
-class_name Player
+
```
Comment (Jamesbarford) [inline Moonshot/player/player.gd:2]: revert

File: Moonshot/player/LedgeWallDetector.gd (hunk 2 above)
Comment (Jamesbarford) [reply Moonshot/player/LedgeWallDetector.gd:18]: @kepler471 ?

**Comment by Jamesbarford** [comment]
This is really nice @wattsit good work 👍

File: Moonshot/player/LedgeWallDetector.gd (hunk 2 above)
Comment (kepler471) [reply Moonshot/player/LedgeWallDetector.gd:18]: This was a fix, will have to keep an eye on it though :)

**Review by kepler471 — APPROVED** [review]

**Pull request merged**